index.js

```diff
@@ -138,3 +138,67 @@ newSightingForm.addEventListener('submit', function (event) {
 /************************** EVENTS PART 2 JS MINI CHALLENGE ******************************/
 
 
+
+// FIRST TRY //
+// const collectionUl = document.querySelector('ul#animals')
+
+// collectionUl.addEventListener('click', function (event){
+
+//     if (event.target.matches('button.like-button')){
+//         const likesPtag = event.target.previousElementSibling
+//         const likes = parseInt(likesPtag.textContent) + 1
+//         likesPtag.textContent = `${likes} Likes`
+//     }
+//     else if (event.target.matches('button.delete-button')){
+//         const animal = event.target.closest('li')
+//         animal.remove()
+//     }
+//     else if (event.target.matches('button.toggle-update-form-button')){
+//         debugger
+//         const animal = event.target.closest('#sighting-description')
+//         animal.style.display = "none"
+//     }
+// })
+
+
+// REVIEW: Different Way to Write the Code //
+
+const animalCollection = document.querySelector('ul#animals')
+
+animalCollection.addEventListener('click', function(event){
+
+    const li = event.target.closest('li')
+
+    if(event.target.matches('button.like-button')) {
+       
+        const likesPtag = li.querySelector('p.likes-display')
+        const currLikes = parseInt(likesPtag.textContent) + 1
+        likesPtag.textContent = `${currLikes} Likes`
+    }
+    else if(event.target.className === 'delete-button'){
+        const li = event.target.closest('li')
+        li.remove()
+    }
+    else if(event.target.matches('button.toggle-update-form-button')){
+        //we have to define the form
+        const form = li.querySelector('form')
+        const updateForm = event.target.nextElementSibling
+        if(form.style.display === 'block'){
+            form.style.display = 'none'
+        }
+        else {
+            form.style.display = 'block'
+        }
+        //form.style.display = form.style.display === 'block' ? 'none' : 'block'
+    }
+})
```

Comment:
Nice! You have an event listener on the entire animals UL element, and based on the class names you are delegating out different logic. Great. First step is getting everything up and running. Awesome. Next, we can refactor for readability and to make our code more reusable.

```
/***** DELIVERABLE 1, 2, & 3 part 1 *****/

const animalsUL = document.querySelector('ul#animals')
animalsUL.addEventListener("click", function(event){
    if (event.target.matches('button.like-button')){
        handleLike(event)
    } else if (event.target.matches('button.delete-button')){
        handleDelete(event)
    } else if (event.target.matches('button.toggle-update-form-button')){
        handleToggleUpdate(event)
    }
})

/***** DELIVERABLE 1 part 2 *****/
function handleLike(e){
    const likesDisplayPTag = e.target.previousElementSibling
    const currentTextContent = likesDisplayPTag.innerText
    let numberOfLikes = parseInt(currentTextContent.split(" ")[0])
    numberOfLikes += 1
    const newTextContent = `${numberOfLikes} Likes`
    likesDisplayPTag.innerText = newTextContent
}

/***** DELIVERABLE 2 part 2 *****/
function handleDelete(e){
    const parentLI = e.target.parentElement
    parentLI.remove()
}

/***** DELIVERABLE 3 part 2 *****/
function handleToggleUpdate(e){
    const updateForm = e.target.nextElementSibling
    toggleHidden(updateForm)
}

/***** DELIVERABLE 3 part 3 *****/
function toggleHidden(element){
    element.style.display === "none" ? element.style.display = "block" : element.style.display = "none"
}
```